lib/explorer/data_frame.ex

```diff
@@ -1756,85 +1761,93 @@ defmodule Explorer.DataFrame do
-  def pivot_longer(df, columns, opts \\ [])
+  def pivot_longer(df, columns_to_pivot, opts \\ [])
 
-  def pivot_longer(df, columns, opts) when is_function(columns),
+  def pivot_longer(df, columns_to_pivot, opts) when is_function(columns_to_pivot),
     do:
       df.names
-      |> Enum.filter(columns)
+      |> Enum.filter(columns_to_pivot)
       |> then(&pivot_longer(df, &1, opts))
 
-  def pivot_longer(df, columns, opts) do
-    opts = Keyword.validate!(opts, value_columns: [], names_to: "variable", values_to: "value")
+  def pivot_longer(df, columns_to_pivot, opts) do
+    opts = Keyword.validate!(opts, keep: [], names_to: "variable", values_to: "value")
     names_to = to_column_name(opts[:names_to])
     values_to = to_column_name(opts[:values_to])
 
-    existing_columns = to_existing_columns(df, columns)
+    columns_to_pivot = to_existing_columns(df, columns_to_pivot)
 
     names = df.names
     dtypes = df.dtypes
 
-    value_columns =
-      case opts[:value_columns] do
+    keep_columns =
+      case opts[:keep] do
         [] ->
-          Enum.filter(names, fn name -> name not in columns end)
+          []
 
         [_ | _] = columns ->
           Enum.each(columns, fn column ->
-            if column in existing_columns,
+            if column in columns_to_pivot,
               do:
                 raise(
                   ArgumentError,
-                  "value columns may not also be ID columns but found #{column} in both"
+                  "columns to keep must not include columns to pivot, but found #{inspect(column)} in both"
                 )
           end)
 
           columns
 
         callback when is_function(callback) ->
-          Enum.filter(names, fn name -> name not in columns && callback.(name) end)
+          Enum.filter(names, fn name -> name not in columns_to_pivot && callback.(name) end)
       end
 
-    value_columns = to_existing_columns(df, value_columns)
+    keep_columns = to_existing_columns(df, keep_columns)
 
     values_dtype =
       dtypes
-      |> Map.take(value_columns)
+      |> Map.take(columns_to_pivot)
       |> Map.values()
       |> Enum.uniq()
       |> case do
         [dtype] ->
           dtype
 
-        [_ | _] ->
+        [_ | _] = dtypes ->
           raise ArgumentError,
-                "value columns may only include one dtype but found multiple dtypes"
+                "columns to pivot must include columns with the same dtype, but found multiple dtypes: #{inspect(dtypes)}"
       end
 
     new_dtypes =
       dtypes
-      |> Map.take(columns)
+      |> Map.take(keep_columns)
       |> Map.put(names_to, :string)
       |> Map.put(values_to, values_dtype)
 
-    out_df = %{df | names: columns ++ [names_to, values_to], dtypes: new_dtypes}
+    out_df = %{df | names: keep_columns ++ [names_to, values_to], dtypes: new_dtypes}
 
-    Shared.apply_impl(df, :pivot_longer, [out_df, value_columns])
+    Shared.apply_impl(df, :pivot_longer, [out_df, columns_to_pivot])
```

Comment:
Let's pass `keep`, `names_to`, and `values_to` to the backend, so it doesn't have to compute it. WDYT? Or at least `names_to` and `values_to`.

Reply:
yeah, makes sense!